datastore/api/snippets.py

```diff
@@ -0,0 +1,823 @@
+        'tags': [
+            'fun',
+            'programming',
+            'learn'
+        ],
+        'collaborators': [
+            'alice',
+            'bob',
+            'charlie'
+        ],
+        'created': datetime.datetime.utcnow()
+    })
+    # [END exploding_queries]
+
+    client.put(task)
+    return task
+
+
+def transactional_update(client):
+    # Create the entities we're going to manipulate
+    account1 = datastore.Entity(client.key('Account'))
+    account1['balance'] = 100
+    account2 = datastore.Entity(client.key('Account'))
+    account2['balance'] = 100
+    client.put_multi([account1, account2])
+
+    # [START transactional_update]
+    def transfer_funds(client, from_key, to_key, amount):
+        with client.transaction():
+            from_account, to_account = client.get_multi([from_key, to_key])
+
+            from_account['balance'] -= amount
+            to_account['balance'] += amount
+
+            client.put_multi([from_account, to_account])
+    # [END transactional_update]
+
+    # [START transactional_retry]
+    for _ in range(5):
+        try:
+            transfer_funds(client, account1.key, account2.key, 50)
+            return
+        except gcloud.exceptions.Conflict:
+            continue
+    # [END transaction_retry]
+
+
+def transactional_get_or_create(client):
+    # [START transactional_get_or_create]
+    with client.transaction():
+        key = client.key('Task', datetime.datetime.utcnow().isoformat())
+
+        task = client.get(key)
+
+        if not task:
+            task = datastore.Entity(key)
+            task.update({
+                'description': 'Example task'
+            })
+            client.put(task)
+
+        return task
+    # [END transactional_get_or_create]
+
+
+def transactional_single_entity_group_read_only(client):
+    client.put_multi([
+        datastore.Entity(key=client.key('TaskList', 'default')),
+        datastore.Entity(key=client.key('TaskList', 'default', 'Task', 1))
+    ])
+
+    # [START transactional_single_entity_group_read_only]
+    with client.transaction():
+        task_list_key = client.key('TaskList', 'default')
+
+        task_list = client.get(task_list_key)
+
+        query = client.query(kind='Task', ancestor=task_list_key)
+        tasks_in_list = list(query.fetch())
+
+        return task_list, tasks_in_list
+    # [END transactional_single_entity_group_read_only]
+
+
+def namespace_run_query(client):
+    # Create an entity in another namespace.
+    task = datastore.Entity(
+        client.key('Task', namespace='namespace-b'))
+    client.put(task)
+
+    # [START namespace_run_query]
+    # All namespaces
+    query = client.query(kind='__namespace__')
+    query.keys_only()
+
+    all_namespaces = [entity.key.id_or_name for entity in query.fetch()]
+
+    # Filtered namespaces
+    query = client.query(kind='__namespace__')
+    query.add_filter(
+        '__key__', '>=', client.key('__namespace__', 'namespace-a'))
```

Comment:
It's still consistent, it just shows how to create this weird key. Consider the key *is weird* and non-trivial I think it's important that we show how to make it.